src/components/ExportHistoryModal.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+    <ModalComponent>
+        <div class="min-w-max flex flex-col">
+            <h2 class="text-xl font-bold mb-4 text-center">Export History</h2>
+            <form>
+                <div>
+                    <label class="mr-2" for="export-format">Export as:</label>
+                    <select class="select" id="export-format" v-model="exportAs">
+                        <option
+                            v-for="option in exportOptions"
+                            :key="option"
+                            :value="option"
+                            v-text="formatField(option)"
+                        />
+                    </select>
+                </div>
+                <div v-if="exportAs === 'history'">
+                    <label v-for="field in fields" :key="field" class="flex gap-2 items-center">
+                        <input
+                            class="checkbox"
+                            type="checkbox"
+                            :value="field"
+                            v-model="selectedFields"
+                            :disabled="field === 'videoId'"
+                        />
+                        <span v-text="formatField(field)" />
+                    </label>
```

Comment:
We likely don't want this to be user selectable, as we want to stick to a standard format between frontend implementations.

We would potentially align to a new format in the distant future to allow inter-compatibility with other frontends once a format is standardized - https://github.com/UniversalPipeWrench/unified-user-data-format/issues/1.

Reply:
I see one problem with this: What if the format we decide on is not what users want? 
For example, if we include watchedAt and currentTime by default this might be a privacy concern for some users. On the other hand, if we decide to leave them out we are throwing out features which I would assume most users want, myself included.
The only solution I can think of is for the standard format to allow these to be optional.

Reply:
I feel `currentTime` is always necessary since otherwise, we can't show any indicators on videos (just the watch history page).

Regarding `watchedAt`, that's a fair point, but we don't know how to sort videos then.

However, I feel this is less of an issue since watch history is stored entirely locally - so, it's unlikely any third party gets access to it in the first place.